vault/resource_jwt_auth_backend.go

```diff
@@ -131,7 +133,176 @@ func jwtAuthBackendResource() *schema.Resource {
+			},
+
+			"oidc_discovery_ca_pem": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The CA certificate or chain of certificates, in PEM format, to use to validate connections to the OIDC Discovery URL. If not set, system certificates are used",
+			},
+
+			"oidc_client_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Client ID used for OIDC",
+			},
+
+			"oidc_client_secret": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				Description: "Client Secret used for OIDC",
+			},
+
+			"jwks_url": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"oidc_discovery_url", "jwt_validation_pubkeys"},
+				Description:   "JWKS URL to use to authenticate signatures. Cannot be used with 'oidc_discovery_url' or 'jwt_validation_pubkeys'.",
+			},
+
+			"jwks_ca_pem": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The CA certificate or chain of certificates, in PEM format, to use to validate connections to the JWKS URL. If not set, system certificates are used.",
+			},
+
+			"jwt_validation_pubkeys": {
+				Type:          schema.TypeList,
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				Optional:      true,
+				ConflictsWith: []string{"jwks_url", "oidc_discovery_url"},
+				Description:   "A list of PEM-encoded public keys to use to authenticate signatures locally. Cannot be used with 'jwks_url' or 'oidc_discovery_url'. ",
+			},
+
+			"bound_issuer": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The value against which to match the iss claim in a JWT",
+			},
+
+			"jwt_supported_algs": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "A list of supported signing algorithms. Defaults to [RS256]",
+			},
+
+			"default_role": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The default role to use if none is provided during login",
+			},
+
+			"accessor": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The accessor of the JWT auth backend",
+			},
+			"provider_config": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "Provider specific handling configuration",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"tune": authMountTuneSchema(),
+		},
+	}
+}
+
+func resourceJwtAuthStateUpgradeV0(rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+
+	// The provider_config schema changed types from TypeMap to
+	// TypeSet, so we require a state upgrade. The values in this
+	// schema are already strings, so we just need to convert the
+	// non-string values.
+	if rawConfig, ok := rawState["provider_config"]; ok {
```

Comment:
Nit: Can you flip the check here to `!ok` and return early so we don't wrap the rest of the logic within an indented block?